lib/AtomsIOPython/src/ase.jl

```diff
@@ -11,26 +11,48 @@ Supported formats:
   - ASE trajectory files
   - [XYZ](https://openbabel.org/wiki/XYZ) and [extxyz](https://github.com/libAtoms/extxyz#extended-xyz-specification-and-parsing-tools) files
 """
-struct AseParser <: AbstractParser end
+@kwdef struct AseParser <: AbstractParser 
+    read::Bool = true
+    guess::Bool = true
+end
 
 
-function AtomsIO.supports_parsing(::AseParser, file; save, trajectory)
+function AtomsIO.supports_parsing(parser::AseParser, file; save, trajectory)
     format = ""
     try
-        format = ase.io.formats.filetype(file; read=!save, guess=false)
+        format = ase.io.formats.filetype(file; read=parser.read, guess=parser.guess)
     catch e
         e isa PyException && return false
         rethrow()
     end
 
     if !(format in ase.io.formats.ioformats)
         return false
-    elseif trajectory
-        # Loading/saving multiple systems is supported only if + in code
-        supports_trajectory = '+' in ase.io.formats.ioformats[format].code
-        return supports_trajectory
-    else
-        return true
+    end
+
+    ioformat = ase.io.formats.ioformats[format]
+    supports_trajectory = '+' in ase.io.formats.ioformats[format].code
+
+    if save # Check whether ASE can write format
```

Comment:
I would structure it as

If save && can write
   Return true if trajectory matches
Elseif !save && can read
   ...
End
Return false